starlite/app.py

```diff
@@ -512,26 +600,52 @@ def _add_node_to_route_map(self, route: BaseRoute) -> RouteMapNode:
         self._configure_route_map_node(route, current_node)
         return current_node
 
-    def _add_route_to_handler_index(self, route: BaseRoute) -> None:
-        """Maps route handler names to urls.
+    def _get_route_handlers(self, route: BaseRoute) -> List["RouteHandlerType"]:
+        """Retrieve handler(s) as a list for given route."""
+        route_handlers: List["RouteHandlerType"] = []
+        if isinstance(route, (WebSocketRoute, ASGIRoute)):
+            route_handlers.append(route.route_handler)
+        else:
+            route_handlers.extend(cast("HTTPRoute", route).route_handlers)
+
+        return route_handlers
+
+    def _get_handler_qualname(self, handler: "RouteHandlerType") -> str:
+        """Retrieves __qualname__ for passed route handler."""
+        if handler.fn and hasattr(handler.fn, "__qualname__"):
+            return handler.fn.__qualname__
+
+        return type(handler.fn).__qualname__
+
+    def _store_handler_to_route_mapping(self, route: BaseRoute) -> None:
+        """Stores mappings:
+
+            * between route handlers and routes;
+            * between route handlers and names.
 
         Args:
             route: A Route instance.
 
         Returns:
             None
         """
-        route_handlers: List["RouteHandlerType"] = []
-        if isinstance(route, (WebSocketRoute, ASGIRoute)):
-            route_handlers.append(route.route_handler)
-        else:
-            route_handlers.extend(cast("HTTPRoute", route).route_handlers)
+        route_handlers = self._get_route_handlers(route)
 
-        for route_handler in filter(lambda x: x.name, route_handlers):
-            name = cast("str", route_handler.name)
-            if name in self._route_handler_index:
+        for handler in route_handlers:
+            qualname = self._get_handler_qualname(handler)
+            self._route_mapping[qualname].append(route)
+
+            name = handler.name
+            if not name:
```

Comment:
so, i think we should revisit this. I would suggest we dont force users to define a handler.name to get this functionality, instead we can do one of the following:

```python
name = handler.name or cast("AnyCallable", handler.fn).__name__
```

This might be quite intuitive to use - the problem is of course some people will have functions with identical names. 

So we can also instead do this:

```python
name = handler.name or cast("AnyCallable", handler.fn).__qualname__
```

Certainly not so intuitive, but will work.

Reply:
@Goldziher so your idea is to use `__qualname__` as default name for handler indexes. Did I get this right?